lib/elastic_apm/timed_worker.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
```

Comment:
Have you considered using a [TimedWorker](http://ruby-concurrency.github.io/concurrent-ruby/Concurrent/TimerTask.html)? I haven't used this myself, but it seems to be created exactly for the purpose of running a task on a scheduled time basis. I think it would be a good design, to have the timer thread separated from the business logic here.

Reply:
@Qard I’m hesitant to add dependencies. I’m not completely sure if it’s really a wasted effort but at least I’m putting it off as much as possible.